rs/sns/root/src/lib.rs

```diff
@@ -704,12 +704,74 @@ impl SnsRootCanister {
         }
     }
 
+    pub async fn maybe_update_framework_canisters_memory_limit(
+        self_ref: &'static LocalKey<RefCell<Self>>,
+        management_canister_client: &impl ManagementCanisterClient,
+    ) {
+        if let Some(canisters) = self_ref.with(|state: &RefCell<SnsRootCanister>| {
+            let mut state = state.borrow_mut();
+            // If we have not updated the framework canisters' memory limit
+            if !state
+                .updated_framework_canisters_memory_limit
+                .unwrap_or_default()
+            {
+                state.updated_framework_canisters_memory_limit = Some(true);
+
+                // Since root doesn't own itself, we can't update root automatically
```

Comment:
So how do we go about Root, does it already have memory limits set to 4 GB by the NNS?

Reply:
Moving this discussion elsewhere as it's not related to the PR.